pkg/storage/sqlcommon/sqlcommon.go

```diff
@@ -690,14 +690,18 @@ func ReadAuthorizationModel(
 
 // IsReady returns true if the connection to the datastore is successful
 // and the datastore has the latest migration applied.
-func IsReady(ctx context.Context, db *sql.DB) (storage.ReadinessStatus, error) {
+func IsReady(ctx context.Context, db *sql.DB, dialect string) (storage.ReadinessStatus, error) {
 	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
 	defer cancel()
 
 	if err := db.PingContext(ctx); err != nil {
 		return storage.ReadinessStatus{}, err
 	}
 
+	if err := goose.SetDialect(dialect); err != nil {
```

Comment:
Instead of doing it here, which will run on every call, can we do it in `sqlcommon.NewDBInfo`?

Reply:
done in [`0af228a`](https://github.com/openfga/openfga/pull/2252/commits/0af228af1e607fa3f62b3f7acc20005e27d02d04)